queries/typescript/scopeTypes.scm

```diff
@@ -0,0 +1,57 @@
+(object) @map
+
+(arrow_function) @anonymousFunction
+(function) @anonymousFunction
+
+; We could also have used an alternation here, but the above style is clean and
+; seems to be more commonly used.
+; [
+;     (arrow_function)
+;     (function)
+; ] @anonymousFunction 
+
+; The following doesn't work because we can't have a `(` before the field name (`key`)
+; (
+;     (object
+;         (pair
+;             (key: (_) @key
+;             .
+;             ":") @key.removalRange
+;         ) @key.searchScope
+;     ) @key.iterationScope
+; )
+
+; As a workaround, we could allow defining the end of an annotation
+(
+    (object
+        (pair
+            key: (_) @key
+            .
+            ":" @key.removalRange.end
+        ) @key.searchScope
+    ) @key.iterationScope
+)
+
+; We could instead tag the delimiter itself.  Gives us a bit more information,
+; so maybe that's better?
+(
+    (object
+        (pair
+            key: (_) @key
+            .
+            ":" @key.delimiter.trailing
```

Comment:
Advantage of labelling the delimiter itself is that we can then target it with the "trailing" modifier (#309)

Tho we can also do this with the approach below where we use `#set!`

But I think we should prob support all 3 tbh, as they could all be useful for different things

Reply:
One question is whether we want to use `@key.delimiter.trailing` here or just `@key.delimiter`